controllers/etcdbackupschedule_controller.go

```diff
@@ -2,32 +2,113 @@ package controllers
 
 import (
 	"context"
+	"sync"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/robfig/cron/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	etcdv1alpha1 "github.com/improbable-eng/etcd-cluster-operator/api/v1alpha1"
 )
 
+type CronScheduler interface {
+	AddFunc(spec string, cmd func()) (cron.EntryID, error)
+}
+
 // EtcdBackupScheduleReconciler reconciles a EtcdBackupSchedule object
 type EtcdBackupScheduleReconciler struct {
 	client.Client
 	Log logr.Logger
+
+	// CronHandler is able to schedule cronjobs to occur at given times.
+	CronHandler CronScheduler
+
+	// Schedules holds a mapping of resources to the object responsible for scheduling the backup to be taken.
+	Schedules map[string]Schedule
+}
+
+type Schedule struct {
+	once      *sync.Once
+	cronEntry cron.EntryID
 }
 
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdbackupschedules,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdbackupschedules/status,verbs=get;update;patch
 
 func (r *EtcdBackupScheduleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("etcdbackupschedule", req.NamespacedName)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	log := r.Log.WithValues("etcdbackupschedule", req.NamespacedName)
+
+	resource := &etcdv1alpha1.EtcdBackupSchedule{}
+	if err := r.Get(ctx, req.NamespacedName, resource); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	schedule, found := r.Schedules[string(resource.UID)]
+	if !found {
+		schedule = Schedule{
+			once: &sync.Once{},
```

Comment:
Why the use of `sync.Once`? If we're in this branch we know we're looking at this schedule for the first time (this execution of the controller). Can't we not use a sync and just execute `r.CronHandler.AddFunc` in here?

Reply:
yep you're right it's not needed. its a leftover from a previous experiment - i've cleaned it up